vimeo/VimeoUpload
# VIM-XXX: Remove Local Quota Check

#### Ticket

https://vimean.atlassian.net/browse/VIM-5824
https://vimean.atlassian.net/browse/VIM-5825
https://vimean.atlassian.net/browse/VIM-5826

#### Issue Summary

We want to remove all local quota checks and instead rely on the backend to tell us when our storage was full so we can perform appropriate measures when this happens. This eliminates a lot of code on the client.

#### Implementation Summary

- Removed all user requests
- Removed all local quota checks
- Killed off a bunch of ugly inheritance
- Renamed a bunch of stuff
    - Removing "quota" from the names
    - Removing "PHAsset" from the names (since we no longer have to distinguish from ALAsset)

#### How to Test

This can be tested in both the VimeoUpload example projects and the app itself (with this PR: https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/1648)

- Make sure everything works (4-step and 2-step)
- In the app, make sure you get the appropriate error when you reach your quota
- Make sure all backgrounding, resuming, and retrying work
